pyccel/ast/cudaext.py

```diff
@@ -174,6 +174,62 @@ def __init__(self, dim=None):
     def dim(self):
         return self._dim
 
+
+class CudaCopy(CudaNewArray):
+    """
+    Represents a call to  cuda.copy for code generation.
+
+    Parameters
+    ----------
+    arg : list, tuple, PythonList
+
+    memory_location : str
+        'host'   the newly created array is allocated on host.
+        'device' the newly created array is allocated on device.
+    
+    is_async: bool
+        Indicates whether the copy is asynchronous or not [Default value: False]
+
+    """
+    __slots__ = ('_arg','_dtype','_precision','_shape','_rank','_order','_memory_location', '_is_async')
+
+    def __init__(self, arg, memory_location, is_async=False):
+
+        if not isinstance(arg, (PythonTuple, PythonList, Variable)):
+            raise TypeError('unknown type of  %s.' % type(arg))
+        
+        # Verify the memory_location of src
+        if arg._memory_location not in ('device', 'host'):
```

Comment:
Why not use the getter of `_memory_location` instead ?

Reply:
I didn't pay attention since I could access it within this class.
It has now been updated 55ac74160903acf9799f1a795cbab1eb035dcf0c